clients/src/main/java/org/apache/kafka/clients/InFlightRequests.java

```diff
@@ -178,7 +177,7 @@ public boolean isEmpty() {
             if (!deque.isEmpty()) {
                 NetworkClient.InFlightRequest request = deque.peekLast();
                 long timeSinceSend = now - request.sendTimeMs;
-                if (timeSinceSend > requestTimeoutMs)
+                if (timeSinceSend > request.requestTimeoutMs)
```

Comment:
Can you explain the reasoning of just looking at the last item in the queue? It may be worth adding a comment.